### ISSUE TITLE
Systray Menu Separators don't respect the submenu placement

### ISSUE THREAD

**Issue by EssGeeEich** [fyne-io/fyne#3642]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

While using the Systray Menu, adding separator items to a submenu results in said separator being added to the "root" systray menu instead of being added to the submenu.

### How to reproduce

Add a Separator Item to a submenu. See snippet below.

### Screenshots

Result:
![7uwgdp7diF](https://user-images.githubusercontent.com/6076005/217676336-7dc111ad-bb4f-4ec1-87d7-407595e1ce4e.png)
Expected:
![wBkLniLVdz](https://user-images.githubusercontent.com/6076005/217676344-65d4555f-677e-436e-9241-a444f0bac1eb.png)

### Example code

```go
func main() {
	a := app.New()

	if desk, ok := a.(desktop.App); ok {

		submenu := fyne.NewMenuItem("Test Submenu", func() {})
		submenu.ChildMenu = fyne.NewMenu(
			"Test Submenu",
			fyne.NewMenuItem("Before Separator", func() {}),
			fyne.NewMenuItemSeparator(),
			fyne.NewMenuItem("After Separator", func() {}),
		)

		var menu *fyne.Menu = fyne.NewMenu("Test", submenu)

		desk.SetSystemTrayMenu(menu)
	}
	a.Run()
}
```

### Fyne version

4744d19

### Go compiler version

1.19.5

### Operating system

Windows

### Operating system version

Windows 10

**Comment by SayHeyD** [comment]
I have a similar issue on MacOS. But instead of the menu separator showing up in the root menu, it doesn't show up at all.

 
<img width="498" alt="Screenshot 2023-02-12 at 00 23 59" src="https://user-images.githubusercontent.com/25387070/218285351-a80c1f26-f4a8-4454-96db-6afd2ae42814.png">

### Code:

```go
func Init() {
	a := app.New()

	keys := key.GetAvailableKeys("")

	var menuItems []*fyne.MenuItem

	for _, ageKey := range keys {

		ageKeyMenu := fyne.NewMenuItem(ageKey.Name, func() {})
		ageKeyMenu.ChildMenu = fyne.NewMenu("key options",
			fyne.NewMenuItem("Encryption and decryption", func() {}),
			fyne.NewMenuItem("Encryption", func() {}),
			fyne.NewMenuItem("Decryption", func() {}),
			fyne.NewMenuItemSeparator(),
			fyne.NewMenuItem("Copy key name", func() {}),
			fyne.NewMenuItem("Copy public key", func() {}),
			fyne.NewMenuItem("Copy private key", func() {}),
			fyne.NewMenuItem("Copy age key", func() {}))

		menuItems = append(menuItems, ageKeyMenu)
	}

	if desk, ok := a.(desktop.App); ok {
		keyMenu := fyne.NewMenuItem("Keys", func() {})
		keyMenu.ChildMenu = fyne.NewMenu("Key menu", menuItems...)

		m := fyne.NewMenu("MainMenu", keyMenu)
		desk.SetSystemTrayMenu(m)
	}

	a.Run()
}
```

### Fyne version
2.3.0

### Go compiler version
1.19.5

### Operating system
MacOS

### Operating system version
Ventura 13.1

### Additional Information
No response

**Comment by andydotxyz** [comment]
Fixed in v2.3.2 coming out this weekend
